### ISSUE TITLE
Intermittent test failure `com.diffplug.spotless.FormatterPropertiesTest`

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/spotless#170]
The following test failure has been showing up occasionally in our builds:

https://travis-ci.org/diffplug/spotless/builds/310652617#L639
```
com.diffplug.spotless.FormatterPropertiesTest > differntPropertyFileTypes FAILED
    java.lang.IllegalArgumentException at FormatterPropertiesTest.java:63
        Caused by: java.io.IOException at FormatterPropertiesTest.java:63
com.diffplug.spotless.FormatterPropertiesTest > multiplePropertyFiles FAILED
    java.lang.IllegalArgumentException at FormatterPropertiesTest.java:77
        Caused by: java.io.IOException at FormatterPropertiesTest.java:77
```

**Comment by nedtwigg** [comment]
As best I can tell, it's because we write a file to disk using `Files.write(file.toPath(), ...)`, then read that file on the the next line with `new FileInputStream(file)`.  Perhaps its because we're mixing File and Path?  I can't reproduce on my mac.

**Comment by fvgh** [comment]
I will have a further look at #170 independently of #172 . But I see no restrictions (path characters, etc). Path should only be more convenient/modern and the switch between "old" File and "new" Path as well as mixing both in case of code heritage, seems to be encouraged.

**Comment by fvgh** [comment]
Issue is fixed with #173.
Problem occurs in the following scenario:
1. DTD is configured with SystemID and external URL in the Formatter properties ([like in the test](https://github.com/diffplug/spotless/blob/master/testlib/src/test/resources/formatter/properties/valid_xml_properties.xml#L2))
2. No connection is available to lookup the DTD

Work-around provided with #173 prevents the Xerces implementation to perform the lookup. For the offending `properties.dtd` later on the Java `Properties` parser uses the private internal `properties.dtd` for validation.

Currently the XMLs supported by the `FormatterProperties` have no XSD/DTD except the `properties.dtd`. 
In case in future more XML formats shall be supported which require XSD/DTD, the  `FormatterProperties` should provide internally the required XSD/DTD. With Java 9 XML catalog support, a slim solution is provided. Hence I propose to skip a "final" fix until it is needed and use the work-around.

**Comment by nedtwigg** [comment]
Thanks Frank!
